### PR TITLE
DATACOUCH-243 - CouchbaseTemplate.doRemove() handle exceptions as the…

### PR DESCRIPTION
**Pull request by chaeyk**
Thank you for proposing a pull request. This template will guide you through the essential steps necessary for a pull request.
Make sure that:

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATACOUCH).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).
- [x] You provide your full name and an email address registered with your GitHub account. If you’re a first-time submitter, make sure you have completed the [Contributor’s License Agreement form](https://support.springsource.com/spring_committer_signup).

… same way with doPersist().

### CODE REVIEW

File: src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java [hunk 1]
```diff
@@ -653,9 +659,14 @@ public Boolean doInBucket() throws InterruptedException, ExecutionException {
     execute(new BucketCallback<Boolean>() {
       @Override
       public Boolean doInBucket() {
-        RawJsonDocument deletedDoc = client.remove(converted.getId(), persistTo, replicateTo
-            , RawJsonDocument.class);
-        return deletedDoc != null;
+        try {
+          RawJsonDocument deletedDoc = client.remove((String) objectToRemove, persistTo, replicateTo,
```
Comment (simonbasle) [inline src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java]: this seems copy-paste from row 643 above. anyway it is wrong, and the integration tests would have shown that (compare again to the code from before the change).

**Comment by simonbasle** [comment]
hey @chaeyk 
the last part of the checklist is the most important! to be accepted, your contribution must be linkable to a signature of the CLA, so you need to make the commit with your name and email address that you used to sign the CLA.

File: src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java (hunk 1 above)
Comment (chaeyk) [inline src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java]: fixed and committed

**Comment by simonbasle** [comment]
Ok great, we're close 👍 

Can you:

 - squash the two commits into a single one? (if you need help with that let me know)
 - **then** make the single commit author information have at least the correct email (`chayek@gmail.com`, it is currently your work email) and preferably full name as well?

The best way to set the correct email address is to configure the git repo accordingly and reset the squashed commit's author field (this will only change the name/email for this repository):

```shell
git config --local user.name "Young-ku Chae"
git config --local chaeyk@gmail.com
git commit --amend --reset-author
````

Then update this pull request:
```shell
git push --force-with-lease
```

**Comment by chaeyk** [comment]
OK, I followed your instructions.
This is first time contributing to open source.
Thank you for your help.

**Comment by simonbasle** [comment]
:+1:

**Pull request merged**